frankDoc/src/main/java/nl/nn/adapterframework/frankdoc/model/FrankDocModel.java

```diff
@@ -77,9 +77,11 @@
 	private final ElementRole.Factory elementRoleFactory = new ElementRole.Factory();
 	private Map<Set<ElementRole.Key>, ElementRoleSet> allElementRoleSets = new HashMap<>();
 	private AttributeEnumFactory attributeEnumFactory = new AttributeEnumFactory();
+	private @Getter String rootClassName;
```

Comment:
Als er een constante is voor 'Module', dan kan die er ook zijn voor 'Configuration', en dan hoeft dit veld er ook niet te zijn.

Reply:
Dat klopt voor productie code, maar niet voor unit tests. In de *.testtarget.* packages zitten class hierchieën waarin de root class anders heet. In die tests willen we ook dat het `<Module>` element wordt toegevoegd en config child is van de root class.

Reply:
In digester-rules.xml staan de volgende rules:

	<rule pattern="IOS-Adaptering" factory="null"/><!-- deprecated, use configuration or module instead -->
	<rule pattern="configuration" factory="null"/>

Ik zou verwachten dat pattern "IOS-Adaptering" zonder probleem weggehaald kan worden. Dit pattern werkt hetzelfde als "Module", maar er is geen rule voor "Module". Als we dat doen, dan wordt "configuration" het enige pattern waar geen "/" in zit. Het root element zit dus al in digester-rules.xml. Ik denk dat de logica die een configuratie parsed deze rule ook nodig heeft. Hoe zou de Digester anders moeten weten wat een `<configuration>` tag inhoudt?

Ik stel voor dat ik IOS-Adaptering weghaal uit digester-rules.xml en dat ik daarna in de Frank!Doc digester-rules.xml gebruik om de class te vinden van het root element van configuraties (nu Configuration).

Reply:
Nee, ik sta niet meer achter mijn laatste opmerking. Je zou uit `digester-rules.xml` kunnen halen dat het root element `<configuration>` is. Je zou daaruit kunnen afleiden wat de simple name van de corresponderende Java class is, maar je hebt geen package name. Ik denk dat het goed is zoals het nu in de PR is.